mythril/support/signatures.py

```diff
@@ -1,44 +1,244 @@
+        :param timeout: online lookup timeout
+        :return: list of matching function signatures
+        """
+        if not sighash.startswith("0x"):
+            sighash = "0x%s" % sighash  # normalize sighash format
+
+        if self.enable_online_lookup and not self.signatures.get(sighash) and sighash not in self.online_lookup_miss and time.time() > self.online_directory_unavailable_until:
+            # online lookup enabled, and signature not in cache, sighash was not a miss earlier, and online directory not down
+            logging.debug("Signatures: performing online lookup for sighash %r" % sighash)
+            try:
+                funcsigs = SignatureDb.lookup_online(sighash, timeout=timeout)  # might return multiple sigs
+                if funcsigs:
+                    # only store if we get at least one result
+                    self.signatures[sighash] = funcsigs
+                else:
+                    # miss
+                    self.online_lookup_miss.add(sighash)
+            except FourByteDirectoryOnlineLookupError as fbdole:
+                self.online_directory_unavailable_until = time.time() + 2 * 60  # wait at least 2 mins to try again
+                logging.warning("online function signature lookup not available. will not try to lookup hash for the next 2 minutes. exception: %r" % fbdole)
+        return self.signatures[sighash]  # raise keyerror
+
+    def __getitem__(self, item):
+        """
+        Provide dict interface Signatures()[sighash]
+        :param item: sighash
+        :return: list of matching signatures
+        """
+        return self.get(sighash=item)
+
+    def import_from_solidity_source(self, code):
+        """
+        Import Function Signatures from solidity source files
+        :param code: solidity source code
+        :return: self
+        """
+        self.signatures.update(SignatureDb.parse_function_signatures_from_solidity_source(code))
+        return self
+
+    @staticmethod
+    def lookup_online(sighash, timeout=None, proxies=None):
+        """
+        Lookup function signatures from 4byte.directory.
+        //tintinweb: the smart-contract-sanctuary project dumps contracts from etherscan.io and feeds them into
+                     4bytes.directory.
+                     https://github.com/tintinweb/smart-contract-sanctuary
+
+        :param sighash: function signature hash as hexstr
+        :param timeout: optional timeout for online lookup
+        :param proxies: optional proxy servers for online lookup
+        :return: a list of matching function signatures for this hash
+        """
+        if not ethereum_input_decoder:
+            return None
+        return list(ethereum_input_decoder.decoder.FourByteDirectory.lookup_signatures(sighash,
+                                                                                       timeout=timeout,
+                                                                                       proxies=proxies))
```

Comment:
i think this will straight away assume proxy is None, can you pass the parameters for proxy? like use the default http_proxy kind of parameters in their respective os. or make your library use the system proxy as default, that would be cool.

Reply:
@norhh `requests` by default uses the system proxy from env if a proxy wasn't explicitly specified. That's why we pass `{}` or `None` (both equivalent) to maintain the default behavior see [github/requests](https://github.com/requests/requests/blob/master/requests/sessions.py#L447).